### PR TITLE
Change connection-string to be object and support making severname, d…

### PR DESCRIPTION
**Pull request by zhanyu2014**
…atabasename visible to customers

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by marstr — CHANGES_REQUESTED** [review]
The descriptions should really be updated to mention that there is a non-trivial string type.

**Comment by marstr** [comment]
@AutorestCI help

**Comment by marstr** [comment]
@AutorestCI rebuild go

**Comment by marstr** [comment]
Ah, another thing that I should point out @zhanyu2014 is that it says your CLA hasn't been signed. I assume you're a Microsoft employee, so you should go to http://opensource and join the appropriate organizations.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json [hunk 1]
```diff
@@ -2393,14 +2393,14 @@
...
-          "$ref": "../datafactory.json#/definitions/SecretBase" 
-        }, 
-        "encryptedCredential": { 
-          "description": "The encrypted credential used for authentication. Credentials are encrypted using the integration runtime credential manager. Type: string (or Expression with resultType string).", 
+        "connectionString": {
+          "description": "An ODBC connection string.",
+          "type": "object"
+        },
+        "encryptedCredential": {
+          "description": "The encrypted credential used for authentication. Credentials are encrypted using the integration runtime credential manager. Type: string (or Expression with resultType string).",
```
[the remark this answers is not in the document]
Comment (zhanyu2014) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json]: Description is updated.

**Comment by marstr** [comment]
Adding the ARM review label before I forget.

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by marstr** [comment]
I'd like to remind you @zhanyu2014 that our License CLA bot still says that you are not appropriately documented as a Microsoft Employee. I cannot merge this PR until your CLA is recognized.

**Comment by marstr** [comment]
Please acknowledge, @zhanyu2014. If I see no response, I'll have to close this PR as inactive.

**Comment by zhanyu2014** [comment]
The related SLA has been validated and passed, related comments in change has been resolved, please help continue to finish this review and merge it. @marstr, @jianghaolu.

**Comment by marstr** [comment]
The only thing remaining now is @ravbhatnagar or another ARM representative to sign off. Thanks for signing the CLA, @zhanyu2014.

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
Ready for me to merge, @zhanyu2014?

**Comment by zhanyu2014** [comment]
@marstr, please help us finish the code merge.

**Pull request merged**